src/custom/state/price/actions.ts

```diff
@@ -9,7 +9,15 @@ export interface ClearQuoteParams {
   chainId: ChainId
 }
 
-export const setLoadingQuote = createAction<boolean>('price/setLoadingQuote')
+export interface SetLoadingQuoteParams {
+  // is it loading
+  loading: boolean
+  // indicator of a necessary hard load
+  // e.g param changes: user changes token, input amt, etc
+  quoteData?: Pick<QuoteInformationObject, 'sellToken' | 'chainId'>
+}
+
+export const setLoadingQuote = createAction<SetLoadingQuoteParams>('price/setLoadingQuote')
```

Comment:
Another alternative I kind of like more is to add two actions `getNewQuote` and `refreshCurrentQuote` which is more semantic. 

- The first action knows it needs to invalidate the previous quote and set the loading flag to true. It just needs  for the token/chainId.
- The second one also doesn't need extra parameters. It just need to set the loading flag to true for the token/chainId.

Reply:
ok, fair enough

Reply:
done here: https://github.com/gnosis/cowswap/pull/738/files#diff-676dbc8b394b3f20c57493545a3c2c7899771246a3bb9c8117dbed3593cf4b79L20